coriolis/tests/worker/rpc/test_server.py

```diff
@@ -119,76 +281,649 @@ def call_exec_task(report_to_conductor=True):
+        )
+
+        mock_get_provider.assert_called_once_with(
+            mock.sentinel.destination_platform_name,
+            constants.PROVIDER_TYPE_DESTINATION_MINION_POOL,
+            None,
+        )
+        mock_validate.assert_called_once_with(
+            mock.sentinel.pool_environment,
+            mock_get_provider.return_value
+            .get_minion_pool_environment_schema.return_value,
+        )
+
+        self.assertEqual(result, (True, None))
+
+        # handle SchemaValidationException
+        mock_validate.side_effect = exception.SchemaValidationException("test")
+        result = self.server.validate_endpoint_destination_minion_pool_options(
+            mock.sentinel.context,
+            mock.sentinel.destination_platform_name,
+            mock.sentinel.pool_environment,
+        )
+
+        self.assertEqual(result, (False, "test"))
+
+    @mock.patch.object(schemas, "validate_value")
+    @mock.patch.object(providers_factory, "get_provider")
+    @mock.patch.object(utils, "get_secret_connection_info")
+    def test_validate_endpoint_connection(
+        self, mock_get_secret, mock_get_provider, mock_validate
+    ):
+        def call_validate_endpoint_connection():
+            return self.server.validate_endpoint_connection(
+                mock.sentinel.context,
+                mock.sentinel.platform_name,
+                mock.sentinel.connection_info,
+            )
+
+        result = call_validate_endpoint_connection()
+
+        mock_get_provider.assert_called_once_with(
+            mock.sentinel.platform_name, constants.PROVIDER_TYPE_ENDPOINT, None
+        )
+        mock_get_secret.assert_called_once_with(
+            mock.sentinel.context, mock.sentinel.connection_info
+        )
+        mock_validate.assert_called_once_with(
+            mock_get_secret.return_value,
+            mock_get_provider.return_value
+            .get_connection_info_schema.return_value,
+        )
+        mock_get_provider.return_value\
+            .validate_connection.assert_called_once_with(
+                mock.sentinel.context, mock_get_secret.return_value
+                )
+
+        self.assertEqual(result, (True, None))
+
+        # handle SchemaValidationException
+        mock_validate.side_effect = exception.SchemaValidationException()
+        result = call_validate_endpoint_connection()
+
+        self.assertEqual(result[0], False)
+
+        # handle ConnectionValidationException
+        mock_validate.side_effect = exception.ConnectionValidationException()
+        result = call_validate_endpoint_connection()
+        self.assertEqual(result[0], False)
+
+        # handle BaseException
+        mock_validate.side_effect = BaseException("test")
+        result = call_validate_endpoint_connection()
+        self.assertEqual(result[0], False)
+
+    @mock.patch.object(providers_factory, "get_provider")
+    @ddt.data(
+        (constants.PROVIDER_TYPE_ENDPOINT, "connection_info_schema"),
```

Comment:
Small nit, but you might as well have the same style of ddt data as the other longer ones. Try:

```python
    @ddt.data(
        (
            constants.PROVIDER_TYPE_ENDPOINT,
            "connection_info_schema"
        ),
        (
            constants.PROVIDER_TYPE_REPLICA_IMPORT,
            "destination_environment_schema",
        ),
        (
            constants.PROVIDER_TYPE_REPLICA_EXPORT,
            "source_environment_schema"
        ),
        (
            constants.PROVIDER_TYPE_SOURCE_MINION_POOL,
            "source_minion_pool_environment_schema",
        ),
        (
            constants.PROVIDER_TYPE_DESTINATION_MINION_POOL,
            "destination_minion_pool_environment_schema",
        ),
    )
```

Reply:
Done